diff --git a/src/jswasi.ts b/src/jswasi.ts
index 5ea6e5d..07acd3a 100644
--- a/src/jswasi.ts
+++ b/src/jswasi.ts
@@ -25,6 +25,7 @@ const CHECKSUM_BINARIES = {
   "/usr/bin/wash": ["resources/wash", "resources/wash.md5"],
 };
 const ALWAYS_FETCH_FILES = {
+  "/usr/bin/init.sh": "resources/init.sh",
   "/etc/motd": "resources/motd.txt",
   "/usr/bin/coreutils": "resources/coreutils",
   "/usr/local/bin/wasibox": "resources/wasibox",
