diff --git a/cmd/age/tui.go b/cmd/age/tui.go
index c0b1b13..ac70dfe 100644
--- a/cmd/age/tui.go
+++ b/cmd/age/tui.go
@@ -14,6 +14,7 @@ package main
 //     No capitalized initials and no periods at the end.
 
 import (
+	"bufio"
 	"bytes"
 	"errors"
 	"fmt"
@@ -121,12 +122,11 @@ func printfToTerminal(format string, v ...interface{}) error {
 
 // readSecret reads a value from the terminal with no echo. The prompt is ephemeral.
 func readSecret(prompt string) (s []byte, err error) {
-	err = withTerminal(func(in, out *os.File) error {
-		fmt.Fprintf(out, "%s ", prompt)
-		defer clearLine(out)
-		s, err = term.ReadPassword(int(in.Fd()))
-		return err
-	})
+	reader := bufio.NewReader(os.Stdin)
+	str, err := reader.ReadString('\n')
+	s = []byte(str)
+	s = bytes.TrimSuffix(s, []byte{10})
+	fmt.Print("\033[F\033[K")
 	return
 }
 
