diff --git a/Cargo.toml b/Cargo.toml
index 82f515b..60704d3 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -7,7 +7,10 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-crossterm = "0.19"
 rust-ini = "0.16"
 lazy_static = "1.4.0"
 rand="0.8.3"
+
+[dependencies.crossterm]
+git = "https://github.com/antmicro/crossterm.git"
+branch = "jswasi"
